go/pkg/zen/validation/validator.go

```diff
@@ -77,25 +82,128 @@ func (v *Validator) Validate(ctx context.Context, r *http.Request) (openapi.BadR
 		},
 	}
 
-	if len(errors) > 0 {
-		err := errors[0]
-		res.Error.Detail = err.Message
-		for _, verr := range err.SchemaValidationErrors {
-			res.Error.Errors = append(res.Error.Errors, openapi.ValidationError{
-				Message:  verr.Reason,
-				Location: verr.Location,
-				Fix:      nil,
-			})
-		}
-		if len(res.Error.Errors) == 0 {
-			res.Error.Errors = append(res.Error.Errors, openapi.ValidationError{
-				Message:  err.Reason,
-				Location: err.ValidationType,
-				Fix:      &err.HowToFix,
-			})
+	// Collect non-nullable errors
+	var nonNullableErrors []*errors.ValidationError
```

Comment:
This is confusing
is a nonNullableError an error that comes from a field being required (not nullable) or is it an error that is not a nullable error 🤔 

can we rename this to be clearer?